internal/declarative/v2/inmemory_rendered.go

```diff
@@ -31,46 +28,24 @@ type InMemoryManifestCache struct {
 	*ttlcache.Cache[string, internal.ManifestResources]
 }
 
-func (c *InMemoryManifestCache) Parse(
-	ctx context.Context, renderer Renderer, obj Object, spec *Spec,
-) (*internal.ManifestResources, error) {
+func (c *InMemoryManifestCache) Parse(spec *Spec,
+) (internal.ManifestResources, error) {
 	file := filepath.Join(ManifestFilePrefix, spec.Path, spec.ManifestName)
 	key := fmt.Sprintf("%s-%s", file, spec.Mode)
 
 	item := c.Cache.Get(key)
 	if item != nil {
 		resources := item.Value()
 
-		copied := &internal.ManifestResources{
```

Comment:
I think this defensive copy was here to ensure that no code can change the cached object (due to a bug or misuse).
Just an example of what is possible: `resources.Items[0].SetDeletionTimestamp(someValue)`.
Now, because you're returning a cache item directly, this protection is disabled. Maybe I am missing something?

Reply:
you are right 👍